### ISSUE TITLE
Create and sync namespace caches on startup

### ISSUE THREAD

**Issue by nrfox** [kiali/kiali#4502]
Namespace caches are currently created and synced when they are first accessed. When there are many namespaces, the Kiali server can take awhile to respond to the first request while it creates and syncs each of the namespace caches. This can be tested by:

1. Creating 300-400 empty namespaces
2. Restarting the kiali pod ensuring no other traffic is going to it
3. As the first request since startup, `curl https://kiali-url/api/mesh/tls`
4. Observe kiali pod logging `Waiting for Kiali cache for ... to sync`

Response times can take 30-60s.

Instead of creating and syncing the Namespace caches when first accessed, Kiali server can create/sync them on startup before accepting any user requests.

**Comment by lucasponce** [comment]
Note that this request is somehow related to https://github.com/kiali/kiali/issues/4382 issue which is working out on https://github.com/kiali/kiali/pull/4479

Main issue here is that the validations logic for tls endpoint doesn't scale well, this issue is located and identified and the strategy we're following is to use Istio Registry in the validations logic, but there are a couple of preliminary steps before to attack the tls endpoint.

Loading all namespaces at startup can be a workaround but is not going to address the real issue here and perhaps it can introduce a slow deployment instead.

**Comment by nrfox** [comment]
> Loading all namespaces at startup can be a workaround but is not going to address the real issue here and perhaps it can introduce a slow deployment instead.

Yes it moves the time spent creating/syncing the caches from "during the first request" to "before the first request". It will make the Kiali deployment slower to become ready initially but I'd argue that is less perceivable to the user than having the first request to Kiali be very slow. In other words, it is part of the setup process for the Kiali server and should be completed before the Kiali server considers itself ready to serve user requests.

> Main issue here is that the validations logic for tls endpoint doesn't scale well

I wouldn't say that's the main issue. It is just circumstance that  the `/mesh/tls` endpoint is both one of the first the UI calls and it uses the kiali cache across namespaces. Regardless of what endpoint the user hits first, if that endpoint uses the kiali cache, then they will need to wait for the caches to be created and synced before the request is completed.

**Comment by lucasponce** [comment]
Not objection to populate the caches on the init phase.

Logic should proabably be placed in this block:
![image](https://user-images.githubusercontent.com/1662329/141941733-dd76cc16-43eb-41b0-a051-5f2565f7b604.png)

And remember that Kiali caches the namespaces that the Kiali ServiceAccount has access (not per user token) but that info is fetched directly from the `business.NamespaceService.GetNamespaces()` service.

**Comment by lucasponce** [comment]
Now that validations are using the Istio Registry, I think this issue can be started.
It should be a short fix, to just populate the informers at the init phase.

I'm going to create a separate issue as I think that the current model has still a performance issue when there are a lot of namespaces, as Kiali needs to create separated informers(s) per namespace and API.

One potential optimization may be to give visibility of this aspect to the user and if the user decides to grant cluster rights to the ServiceAccount, Kiali can highly simplify the informers' creation, but I'll describe this better in a separate issue.

This issue can be used to just change and populate caches on startup instead of on demand.
